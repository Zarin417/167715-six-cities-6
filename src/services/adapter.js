export const adaptOffersData = (data) => {
  const adaptedData = ({
    ...data,
    isFavorite: data.is_favorite,
    isPremium: data.is_premium,
    previewImage: data.preview_image,
    maxAdults: data.max_adults,
    host: {
      ...data.host,
      avatarUrl: data.host.avatar_url,
      isPro: data.host.is_pro
    }
  });

  delete adaptedData.is_favorite;
  delete adaptedData.is_premium;
  delete adaptedData.preview_image;
  delete adaptedData.max_adults;
  delete adaptedData.host.avatar_url;
  delete adaptedData.host.is_pro;

  return adaptedData;
};

export const adaptReviewsData = (data) => {
  const adaptedData = {
    ...data,
    user: {
      ...data.user,
      avatarUrl: data.user.avatar_url,
      isPro: data.user.is_pro
    }
  };

  delete adaptedData.user.avatar_url;
  delete adaptedData.user.is_pro;

  return adaptedData;
};
