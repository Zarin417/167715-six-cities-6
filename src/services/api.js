import axios from "axios";

const TIMEOUT = 5000;
const BASE_URL = `https://6.react.pages.academy/six-cities`;

const statusCode = {
  UNAUTHORIZED: 401,
};

export const createAPI = (unAuthorized) => {
  const api = axios.create({
    baseURL: BASE_URL,
    timeout: TIMEOUT,
    withCredentials: true,
  });

  const onSuccess = (response) => response;

  const onError = (error) => {
    const {response} = error;

    if (response.status === statusCode.UNAUTHORIZED) {
      unAuthorized();
    }

    throw error;
  };

  axios.interceptors.response.use(onSuccess, onError);

  return api;
};
