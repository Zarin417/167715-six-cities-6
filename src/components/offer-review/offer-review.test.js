import React from "react";
import {render} from "@testing-library/react";
import configureStore from "redux-mock-store";
import {Provider} from "react-redux";
import OfferReview from "./offer-review";
import {createAPI} from "../../services/api";
import thunk from "redux-thunk";

const api = createAPI(() => {});
const mockStore = configureStore([thunk.withExtraArgument(api)]);

describe(`Test 'OfferReview'`, () => {
  it(`OfferReview should render empty review list if reviews are missing`, () => {
    const store = mockStore({
      REVIEW: {isReviewsLoaded: true, placeReviews: []},
      USER: {isLoggedIn: true}
    });

    const {container} = render(
        <Provider store={store}>
          <OfferReview
            placeId={`24`}
          />
        </Provider>
    );

    expect(container).toMatchSnapshot();
  });

  it(`OfferReview should render 'Loader' if data is loading`, () => {
    const store = mockStore({
      REVIEW: {isReviewsLoaded: false},
      USER: {isLoggedIn: true}
    });

    const {container} = render(
        <Provider store={store}>
          <OfferReview
            placeId={`44`}
          />
        </Provider>
    );

    expect(container).toMatchSnapshot();
  });

  it(`OfferReview should render correctly`, () => {
    const store = mockStore({
      USER: {isLoggedIn: true},
      REVIEW: {
        isReviewsLoaded: true,
        placeReviews: [
          {
            comment: ``,
            date: `2020-04-23T08:04:28.647Z`,
            id: 2,
            rating: 4,
            user: {
              avatarUrl: ``,
              id: 8,
              isPro: true,
              name: `Aretha`,
            }
          }
        ]
      }
    });

    const {container} = render(
        <Provider store={store}>
          <OfferReview placeId={`53`}/>
        </Provider>
    );

    expect(container).toMatchSnapshot();
  });
});
