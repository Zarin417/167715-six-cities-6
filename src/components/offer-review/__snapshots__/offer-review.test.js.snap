// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test 'OfferReview' OfferReview should render 'Loader' if data is loading 1`] = `
<div>
  <div
    class="loader"
  >
    <div
      class="loader__dual-ring"
    />
    <p>
      Loading...
    </p>
  </div>
</div>
`;

exports[`Test 'OfferReview' OfferReview should render correctly 1`] = `
<div>
  <section
    class="property__reviews reviews"
  >
    <h2
      class="reviews__title"
    >
      Reviews ·
      <span
        class="reviews__amount"
      >
        1
      </span>
    </h2>
    <ul
      class="reviews__list"
    >
      <li
        class="reviews__item"
      >
        <div
          class="reviews__user user"
        >
          <div
            class="reviews__avatar-wrapper user__avatar-wrapper"
          >
            <img
              alt="Reviews avatar"
              class="reviews__avatar user__avatar"
              height="54"
              src=""
              width="54"
            />
          </div>
          <span
            class="reviews__user-name"
          >
            Aretha
          </span>
          <span
            class="property__user-status"
          >
            Pro
          </span>
        </div>
        <div
          class="reviews__info"
        >
          <div
            class="reviews__rating rating"
          >
            <div
              class="reviews__stars rating__stars"
            >
              <span
                style="width: 80%;"
              />
              <span
                class="visually-hidden"
              >
                Rating
              </span>
            </div>
          </div>
          <p
            class="reviews__text"
          />
          <time
            class="reviews__time"
            datetime="2020-04-23"
          >
            April 23
          </time>
        </div>
      </li>
    </ul>
    <form
      action="#"
      class="reviews__form form"
      method="post"
    >
      <label
        class="reviews__label form__label"
        for="review"
      >
        Your review
      </label>
      <div
        class="reviews__rating-form form__rating"
      >
        <input
          class="form__rating-input visually-hidden"
          id="5-stars"
          name="rating"
          type="radio"
          value="5"
        />
        <label
          class="reviews__rating-label form__rating-label"
          for="5-stars"
          title="perfect"
        >
          <svg
            class="form__star-image"
            height="33"
            width="37"
          >
            <use
              xlink:href="#icon-star"
            />
          </svg>
        </label>
        <input
          class="form__rating-input visually-hidden"
          id="4-stars"
          name="rating"
          type="radio"
          value="4"
        />
        <label
          class="reviews__rating-label form__rating-label"
          for="4-stars"
          title="good"
        >
          <svg
            class="form__star-image"
            height="33"
            width="37"
          >
            <use
              xlink:href="#icon-star"
            />
          </svg>
        </label>
        <input
          class="form__rating-input visually-hidden"
          id="3-stars"
          name="rating"
          type="radio"
          value="3"
        />
        <label
          class="reviews__rating-label form__rating-label"
          for="3-stars"
          title="not-bad"
        >
          <svg
            class="form__star-image"
            height="33"
            width="37"
          >
            <use
              xlink:href="#icon-star"
            />
          </svg>
        </label>
        <input
          class="form__rating-input visually-hidden"
          id="2-stars"
          name="rating"
          type="radio"
          value="2"
        />
        <label
          class="reviews__rating-label form__rating-label"
          for="2-stars"
          title="badly"
        >
          <svg
            class="form__star-image"
            height="33"
            width="37"
          >
            <use
              xlink:href="#icon-star"
            />
          </svg>
        </label>
        <input
          class="form__rating-input visually-hidden"
          id="1-stars"
          name="rating"
          type="radio"
          value="1"
        />
        <label
          class="reviews__rating-label form__rating-label"
          for="1-stars"
          title="terribly"
        >
          <svg
            class="form__star-image"
            height="33"
            width="37"
          >
            <use
              xlink:href="#icon-star"
            />
          </svg>
        </label>
      </div>
      <textarea
        class="reviews__textarea form__textarea"
        id="comment"
        maxlength="300"
        name="comment"
        placeholder="Tell how was your stay, what you like and what can be improved"
      />
      <div
        class="reviews__button-wrapper"
      >
        <p
          class="reviews__help"
        >
          To submit review please make sure to set
          <span
            class="reviews__star"
          >
            rating
          </span>
          and describe your stay with at least
          <b
            class="reviews__text-amount"
          >
            50 characters
          </b>
          .
        </p>
        <button
          class="reviews__submit form__submit button"
          disabled=""
          type="submit"
        >
          Submit
        </button>
      </div>
    </form>
  </section>
</div>
`;

exports[`Test 'OfferReview' OfferReview should render empty review list if reviews are missing 1`] = `
<div>
  <section
    class="property__reviews reviews"
  >
    <h2
      class="reviews__title"
    >
      Reviews ·
      <span
        class="reviews__amount"
      >
        0
      </span>
    </h2>
    <form
      action="#"
      class="reviews__form form"
      method="post"
    >
      <label
        class="reviews__label form__label"
        for="review"
      >
        Your review
      </label>
      <div
        class="reviews__rating-form form__rating"
      >
        <input
          class="form__rating-input visually-hidden"
          id="5-stars"
          name="rating"
          type="radio"
          value="5"
        />
        <label
          class="reviews__rating-label form__rating-label"
          for="5-stars"
          title="perfect"
        >
          <svg
            class="form__star-image"
            height="33"
            width="37"
          >
            <use
              xlink:href="#icon-star"
            />
          </svg>
        </label>
        <input
          class="form__rating-input visually-hidden"
          id="4-stars"
          name="rating"
          type="radio"
          value="4"
        />
        <label
          class="reviews__rating-label form__rating-label"
          for="4-stars"
          title="good"
        >
          <svg
            class="form__star-image"
            height="33"
            width="37"
          >
            <use
              xlink:href="#icon-star"
            />
          </svg>
        </label>
        <input
          class="form__rating-input visually-hidden"
          id="3-stars"
          name="rating"
          type="radio"
          value="3"
        />
        <label
          class="reviews__rating-label form__rating-label"
          for="3-stars"
          title="not-bad"
        >
          <svg
            class="form__star-image"
            height="33"
            width="37"
          >
            <use
              xlink:href="#icon-star"
            />
          </svg>
        </label>
        <input
          class="form__rating-input visually-hidden"
          id="2-stars"
          name="rating"
          type="radio"
          value="2"
        />
        <label
          class="reviews__rating-label form__rating-label"
          for="2-stars"
          title="badly"
        >
          <svg
            class="form__star-image"
            height="33"
            width="37"
          >
            <use
              xlink:href="#icon-star"
            />
          </svg>
        </label>
        <input
          class="form__rating-input visually-hidden"
          id="1-stars"
          name="rating"
          type="radio"
          value="1"
        />
        <label
          class="reviews__rating-label form__rating-label"
          for="1-stars"
          title="terribly"
        >
          <svg
            class="form__star-image"
            height="33"
            width="37"
          >
            <use
              xlink:href="#icon-star"
            />
          </svg>
        </label>
      </div>
      <textarea
        class="reviews__textarea form__textarea"
        id="comment"
        maxlength="300"
        name="comment"
        placeholder="Tell how was your stay, what you like and what can be improved"
      />
      <div
        class="reviews__button-wrapper"
      >
        <p
          class="reviews__help"
        >
          To submit review please make sure to set
          <span
            class="reviews__star"
          >
            rating
          </span>
          and describe your stay with at least
          <b
            class="reviews__text-amount"
          >
            50 characters
          </b>
          .
        </p>
        <button
          class="reviews__submit form__submit button"
          disabled=""
          type="submit"
        >
          Submit
        </button>
      </div>
    </form>
  </section>
</div>
`;
